src/TestProblem/Hydro/ClusterMerger/Flu_ResetByUser_ClusterMerger.cpp

```diff
@@ -0,0 +1,727 @@
+//                2. Invoked using the function pointer "Flu_ResetByUser_API_Ptr"
+//                   --> This function pointer is reset by Init_TestProb_Hydro_ClusterMerger()
+//                3. Currently does not work with "OPT__OVERLAP_MPI"
+//                4. Invoke Flu_ResetByUser_Func_ClusterMerger() directly
+//
+// Parameter   :  lv    : Target refinement level
+//                FluSg : Target fluid sandglass
+//                MagSg   : Target B field sandglass
+//                TimeNew : Current physical time (system has been updated from TimeOld to TimeNew in EvolveLevel())
+//                dt      : Time interval to advance solution (can be different from TimeNew-TimeOld in COMOVING)
+//-------------------------------------------------------------------------------------------------------
+void Flu_ResetByUser_API_ClusterMerger( const int lv, const int FluSg, const int MagSg, const double TimeNew, const double dt )
+{
+
+   const bool    CurrentMaxLv = (  NPatchTotal[lv] > 0  &&  ( lv == MAX_LEVEL  ||  NPatchTotal[lv+1] == 0 )  );
+   const double  dh           = amr->dh[lv];
+   const real    dv           = CUBE(dh);
+#  if ( MODEL == HYDRO  &&  !defined SRHD )
+   const real    Gamma_m1     = GAMMA - (real)1.0;
+   const real   _Gamma_m1     = (real)1.0 / Gamma_m1;
+#  endif
+
+// (1) get the BH position and velocity and adjust them if needed
+   bool AdjustPosNow = false;
+   bool AdjustVelNow = false;
+
+// only adjust the BHs on the current maximum level
+   if ( CurrentMaxLv  &&  AdjustCount < int(TimeNew/AdjustPeriod) )
+   {
+      if ( AdjustBHPos == true )   AdjustPosNow = true;
+      if ( AdjustBHVel == true )   AdjustVelNow = true;
+      AdjustCount += 1;
+   } // if ( CurrentMaxLv  &&  AdjustCount < int(TimeNew/AdjustPeriod))
+   GetClusterCenter( lv, AdjustPosNow, AdjustVelNow, BH_Pos, ClusterCen, BH_Vel );
+
+
+// (2) decide whether to merge BHs
+   double RelativeBHPos[3] = { BH_Pos[0][0]-BH_Pos[1][0], BH_Pos[0][1]-BH_Pos[1][1], BH_Pos[0][2]-BH_Pos[1][2] };
+   double RelativeBHVel[3] = { BH_Vel[0][0]-BH_Vel[1][0], BH_Vel[0][1]-BH_Vel[1][1], BH_Vel[0][2]-BH_Vel[1][2] };
+   double AbsRelPos        = sqrt( SQR(RelativeBHPos[0]) + SQR(RelativeBHPos[1]) + SQR(RelativeBHPos[2]) );
+   double AbsRelVel        = sqrt( SQR(RelativeBHVel[0]) + SQR(RelativeBHVel[1]) + SQR(RelativeBHVel[2]) );
+   double escape_vel;
+   double soften           = amr->dh[MAX_LEVEL];
+   if ( AbsRelPos > soften )
+   {
+      escape_vel = sqrt( 2 * NEWTON_G * (Bondi_MassBH1+Bondi_MassBH2) / AbsRelPos );
+   }
+   else
+   {
+      escape_vel = sqrt( 2 * NEWTON_G * (Bondi_MassBH1+Bondi_MassBH2) / soften );
+   } // if ( AbsRelPos > soften ) ... else ...
+
+// merge the two BHs if they are located within R_acc, and the relative velocity is small enough
+   if ( Merger_Coll_NumHalos == 2 )
+   {
+      if ( AbsRelPos < R_acc  &&  AbsRelVel < 3*escape_vel )
+      {
+         Merger_Coll_NumHalos -= 1;
```

Comment:
This is a reminder to myself that we should not update this value (because it's initially a runtime parameter) but instead create `Merger_Coll_NumBHs` and use and update that.